gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/MySqlDagManagementStateStoreTest.java

```diff
@@ -90,41 +103,40 @@ public void testAddDag() throws Exception {
     DagManager.DagId dagId2 = DagManagerUtils.generateDagId(dag2);
     DagNodeId dagNodeId = DagManagerUtils.calcJobId(dagNode.getValue().getJobSpec().getConfig());
 
-    this.dagManagementStateStore.checkpointDag(dag);
-    this.dagManagementStateStore.checkpointDag(dag2);
+    this.dagManagementStateStore.addDag(dag);
+    this.dagManagementStateStore.addDag(dag2);
     this.dagManagementStateStore.addDagNodeState(dagNode, dagId);
     this.dagManagementStateStore.addDagNodeState(dagNode2, dagId);
     this.dagManagementStateStore.addDagNodeState(dagNode3, dagId2);
 
-    Assert.assertTrue(this.dagManagementStateStore.containsDag(dagId));
-    Assert.assertEquals(dag.toString(), this.dagManagementStateStore.getDag(dagId).get().toString());
+    Assert.assertTrue(compareLists(dag.getNodes(), this.dagManagementStateStore.getDag(dagId).get().getNodes()));
     Assert.assertEquals(dagNode, this.dagManagementStateStore.getDagNodeWithJobStatus(dagNodeId).getLeft().get());
 
     Set<Dag.DagNode<JobExecutionPlan>> dagNodes = this.dagManagementStateStore.getDagNodes(dagId);
     Assert.assertEquals(2, dagNodes.size());
     Assert.assertTrue(dagNodes.contains(dagNode));
     Assert.assertTrue(dagNodes.contains(dagNode2));
 
-    this.dagManagementStateStore.deleteDagNodeState(dagId, dagNode);
-    Assert.assertFalse(this.dagManagementStateStore.getDagNodes(dagId).contains(dagNode));
-    Assert.assertTrue(this.dagManagementStateStore.getDagNodes(dagId).contains(dagNode2));
-    Assert.assertTrue(this.dagManagementStateStore.getDagNodes(dagId2).contains(dagNode3));
-
     // test to verify that adding a new dag node with the same dag node id (defined by the jobSpec) replaces the existing one
-    Assert.assertEquals(this.dagManagementStateStore.getDagNodes(dagId).size(), 1);
```

Comment:
do we not want to assert the size before adding the duplicate?

Reply:
we did that right above.... somewhat hidden because of the diffs in github UI